smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/ack/RabbitMQAckHandler.java

```diff
@@ -0,0 +1,35 @@
+package io.smallrye.reactive.messaging.rabbitmq.ack;
+
+import java.util.concurrent.CompletionStage;
+
+import io.smallrye.reactive.messaging.rabbitmq.IncomingRabbitMQMessage;
+import io.vertx.mutiny.core.Context;
+
+/**
+ * Implemented to provide message acknowledgement strategies.
+ */
+public interface RabbitMQAckHandler {
+
+    enum Strategy {
+        /**
+         * Do no explicitly ack as it will be handled by RabbitMQ itself.
```

Comment:
For my own curiosity, when is it nacked? On reception? If so that may introduce message loss.

Reply:
Yes in auto-ack mode the RabbitMQ broker regards successful delivery as an implied ACK; it provides faster throughput of messages but only in situations where the consumer is prepared to accept that some messages may be lost (e.g. you might use it for blog comments but not for bank transfers).

Reply:
That's a great example! 

This needs to be documented. Kafka has the same "issue" if you do anything async.